calico-enterprise_versioned_docs/version-3.15/_includes/release-notes/_v3.15.2-release-notes.mdx

```diff
@@ -0,0 +1,38 @@
+22 Mar 2023
+
+### Improvements
+
+- Added tolerations to CSI Driver so it can be scheduled on all nodes.
+- Adding graceful shutdown to Typha improves the "happy case" for upgrade, ensuring that shutting down a Typha instance doesn't cause a thundering herd of `calico-node` subcomponent restarts. Instead, clients are disconnected slowly over a configured interval, thus limiting the rate of `calico-node` subcomponent restarts.
```

Comment:
I think this is a v3.15.1 note.

Reply:
The first one is from your post on eng-eng. It sounded like an improvement, not a bug so I put it there. The second one is from 3.15.1, but Matt said patch RN are cumulative, not discrete.